internal/inflector.go

```diff
@@ -0,0 +1,103 @@
+package internal
+
+import (
+	"io/ioutil"
+
+	"github.com/gedex/inflector"
+	"github.com/jinzhu/inflection"
+	"gopkg.in/yaml.v2"
+)
+
+type Inflector interface {
+	Singularize(string) string
+	Pluralize(string) string
+}
+
+type DefaultInflector struct{}
+type RuleInflector struct{}
+
+func (i *DefaultInflector) Singularize(s string) string {
+	return inflector.Singularize(s)
+}
+func (i *DefaultInflector) Pluralize(s string) string {
+	return inflector.Pluralize(s)
+}
+
+func (i *RuleInflector) Singularize(s string) string {
+	return inflection.Singular(s)
+}
+func (i *RuleInflector) Pluralize(s string) string {
+	return inflection.Plural(s)
+}
+
+func NewInflector(ruleFile string) (Inflector, error) {
+	if ruleFile == "" {
+		return &DefaultInflector{}, nil
+	}
+	err := registerRule(ruleFile)
+	if err != nil {
+		return nil, err
+	}
+	return &RuleInflector{}, nil
+}
+
+type rule struct {
+	UnCountable []string       `yaml:"uncountable"`
```

Comment:
I think this can be replaced by irregular. of course even irregular can be replaced by plural and singular pair though.
But I think `uncountable` is not always correct for the name that does not change for plural and singular.
So it's better to just use `irregular` even for those cases.

Honestly, I think plural and singular pair seem complicated for our use case.
It may enough that every additional rules are just represented as irregular. But if we do that, I want to use a different name for that instead of irregular.

Reply:
fixed at b94b3ea
change inflection rule file format like
```
- singular: person
  plural: people
- singular: live
  plural: lives
```